### PR TITLE
Move handling RRule from lib to demo

### PR DESCRIPTION
**Pull request by Fafruch**
RRule is no longer being shown and copy button render by package itself.

onMount handler added to generate default rrule after RRuleGenerator is being mounted. It can be used for example externally as default RRule value being caught without need to change default, initial rrule setting. 
If user wanted to copy initial rrule, it wouldn't be possible, as no onChange would not be triggered and no rrule could be passed externally.

How demo looks now:

screenshot 2017-12-03 at 06 09 14 pm

### CODE REVIEW

**Comment by Fafruch** [comment]
Probably we can get rid of onMount() function and only have:
```
componentDidMount() {
    this.props.onChange(this.state.rrule);
  }
``` 
So 'fake' change will be triggered on component mount.

File: src/demo/App.js [hunk 1]
```diff
@@ -1,40 +1,107 @@
...
-    </div>
-
-    <div className="app">
-      <ReactRRuleGenerator />
-    </div>
-  </div>
-);
+class App extends Component {
+  state = {
+    rrule: '',
```
Comment (tb) [inline src/demo/App.js]: Set the default rrule string here.

File: src/demo/App.js [hunk 2]
```diff
@@ -1,40 +1,107 @@
...
+          Recurence rules generator form built with React
+        </div>
+
+
+        <div className="app container">
+          <h5><strong>{'<RRuleGenerator />'}</strong></h5>
+
+          <ReactRRuleGenerator
+            onMount={defaultRRule => this.setState({ rrule: defaultRRule })}
+            onChange={newRRule => this.handleChange(newRRule)}
```
Comment (tb) [inline src/demo/App.js]: I done see why onMount would sent value to container. Its the other way round - the container should send value down to the editor i.e. like this:

```
<ReactRRuleGenerator
  value={this.state. rrule}
  onChange={this.handleChange}
/>
```

File: src/demo/App.js [hunk 3]
```diff
@@ -1,40 +1,107 @@
...
-    </div>
-  </div>
-);
+class App extends Component {
+  state = {
+    rrule: '',
+    isCopied: false,
+  };
+
+  handleChange = (newRRule) => {
```
Comment (tb) [inline src/demo/App.js]: This is correct and expected. With it you do not need `onChange={newRRule => this.handleChange(newRRule)}`. All you need is `onChange={this.handleChange}` Please check the template.

File: src/lib/components/ReactRRuleGenerator.js [hunk 4]
```diff
@@ -20,46 +23,32 @@ class ReactRRuleGenerator extends Component {
 
       this.props.onChange(rrule);
 
-      return { data: newData, isCopied: false, rrule };
+      return { data: newData, rrule };
```
Comment (tb) [inline src/lib/components/ReactRRuleGenerator.js]: Its the role of container to send state down to ReactRRuleGenerator via `this.props.value`.  You can convert it into local state, thats ok and I would suggest to do it within `componentWillReceiveProps`, see http://www.react.express/lifecycle_api for details.

Check source of  react-select https://github.com/JedWatson/react-select/blob/0d0ccd3acc4efdba0b32d711a6c16809a6d779b4/src/Select.js#L93

File: src/demo/App.js (hunk 2 above)
Comment (Fafruch) [reply src/demo/App.js]: Before I go any further, should default RRule string has an influence on the default view - it means that to configure the view to be adequate to RRule provided and match it 1:1?

Comment (tb) [reply src/demo/App.js]: I think flow should be like this:
- component gests rrule value via `value` prop
- if that value is different the its internal state `rrule` then it: updates it and generates the state based on it

Then, when the rule is changed, you can do the other way round:
- first update internal state (including internal `rrule`)
- then call `onChange` with new `rrule`
This rrule will get to component again via prop, but you will be able to ignore it using  `shouldComponentUpdate` that will compare the `value` and `state.rrule`

**Pull request merged**